File: Games-Store-BE/src/main/java/com/gamesstorebe/service/WishlistService.java
package com.gamesstorebe.service;

import com.gamesstorebe.dto.WishlistDTO;
import com.gamesstorebe.entity.Product;
import com.gamesstorebe.entity.Wishlist;

import java.util.List;

public interface WishlistService {
    List<WishlistDTO> getWishlistByUser(String user);
    List<WishlistDTO> addProductIntoWishList (int productId, String email);
    List<WishlistDTO> removeProductFromWishList (int productId, String email);

    WishlistDTO getWishlistByEmailProductID (int productId, String email);

}

File: Games-Store-FE/src/app/core/service/wishlist.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { BehaviorSubject, Observable } from "rxjs";
import { environment } from "src/environments/environment.development";

const API = environment.apiUrl;
@Injectable({
    providedIn: 'root'
})


export class WishlistService {
    constructor(
        private httpClient: HttpClient
    ) { }
    wishlistChanged: BehaviorSubject<any> = new BehaviorSubject<any>([]);


    getWishlistByUserEmail(userEmail: string): Observable<any> {
        return this.httpClient.get(API + "/wishlist/get-wishlist-by-email?email=" + userEmail);
    }
    addProductIntoWishlist(email: String, productId: number): Observable<any> {

        return this.httpClient.put(API + "/wishlist/add-product-to-wishlist?email=" + email + "&productId=" + productId, null);
    }
    removeProductFromWishlist(email: String, productId: number): Observable<any> {

        return this.httpClient.delete(API + "/wishlist/delete-product-from-wishlist?email=" + email + "&productId=" + productId);
    }

    updateCart() {
        this.wishlistChanged.next([]);
    }
}